QModManager/API/QModAPI.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+namespace QModManager.API
+{
+    public partial class QModAPI : IQModAPI
+    {
+        public static IQModAPI Main { get; } = new QModAPI();
+        private QModAPI() { }
+
+        internal static List<Assembly> ErroredMods = new List<Assembly>();
+
+        public static void MarkAsErrored(Assembly modAssembly = null)
+            => Main.MarkAsErrored(modAssembly);
+
+        public static ReadOnlyCollection<IQMod> GetAllMods(bool includeUnloaded = false, bool includeErrored = false)
+            => Main.GetAllMods(includeUnloaded, includeErrored);
+
+        public static IQMod GetMyMod()
+            => Main.GetMyMod();
+        public static IQMod GetMod(Assembly modAssembly, bool includeUnloaded = false, bool includeErrored = false)
+            => Main.GetMod(modAssembly, includeUnloaded, includeErrored);
+        public static IQMod GetMod(string id, bool includeUnloaded = false, bool includeErrored = false)
+            => Main.GetMod(id, includeUnloaded, includeErrored);
+
+        public static bool ModPresent(Assembly modAssembly, bool includeUnloaded = false, bool includeErrored = false)
+            => Main.ModPresent(modAssembly, includeUnloaded, includeErrored);
+        public static bool ModPresent(string id, bool includeUnloaded = false, bool includeErrored = false)
+            => Main.ModPresent(id, includeUnloaded, includeErrored);
+
+        #region Non-static
+
+        void IQModAPI.MarkAsErrored(Assembly modAssembly = null)
+        {
+            modAssembly = modAssembly ?? Assembly.GetCallingAssembly();
```

Comment:
`Assembly.GetCallingAssembly()` should be replaced with an utility method which gets the calling assembly by stacktrace

Reply:
Pinch from [SMLHelper](https://github.com/SMLHelper/SMLHelper/blob/DevMar2019/SMLHelper/Utility/ReflectionHelper.cs#L173)

Reply:
I know, I was leaving a comment for myself 😅